test/ectest.c

```diff
@@ -1598,8 +1598,40 @@ static int parameter_test(void)
     OPENSSL_free(buf);
     return r;
 }
+
+static int check_ec_key_field_public_range(void)
+{
+    int ret = 0;
+    const EC_POINT *pub = NULL;
+    const EC_GROUP *group = NULL;
+    BIGNUM *x = NULL, *y = NULL;
+    EC_KEY *key = NULL;
+
+    ret = TEST_ptr(x = BN_new())
+          && TEST_ptr(y = BN_new())
+          && TEST_ptr(key = EC_KEY_new_by_curve_name(NID_sect163k1))
```

Comment:
should we make this test run over all the supported curves instead of a single one?

Of course, you would then do the `BN_GF2m_add(x, x, field)` step only for binary curves, and could, by the same logic, do `BN_add(x, x, field)` for the prime curves.

Reply:
Changed to do all curves